### PR TITLE
CAMEL-14805: Component - Use doInit for wiring instead of doStart

### PR DESCRIPTION
**Pull request by zhfeng**
* camel-avro-rpc
* camel-browse
* camel-cxf
* camel-dataset
* camel-distruptor
* camel-elsql
* camel-flink
* camel-fop

### CODE REVIEW

File: components/camel-elsql/src/main/java/org/apache/camel/component/elsql/ElsqlEndpoint.java [hunk 1]
```diff
@@ -121,16 +123,39 @@ protected void doStart() throws Exception {
...
-            list.add(url);
+
+        StreamSupport.stream(it.spliterator(), false)
+                .filter(predicate)
+                .forEach(path -> {
+                    try {
+                        final URL url = ResourceHelper.resolveMandatoryResourceAsUrl(getCamelContext().getClassResolver(), path.toString());
+                        list.add(url);
+                    } catch (Exception e) {
+
```
Comment (davsclaus) [inline components/camel-elsql/src/main/java/org/apache/camel/component/elsql/ElsqlEndpoint.java:151]: This is wrong as you ignore thrown exception - I suggest to NOT use streaming API as its also TERRIBLE sometimes as in this example being overly complex and causing you to do try .. catch and ignore exceptions.

Comment (zhfeng) [reply components/camel-elsql/src/main/java/org/apache/camel/component/elsql/ElsqlEndpoint.java:151]: OK, I see and will update to use the iterable as usual.

Comment (davsclaus) [reply components/camel-elsql/src/main/java/org/apache/camel/component/elsql/ElsqlEndpoint.java:151]: Thanks

File: components/camel-elsql/src/main/java/org/apache/camel/component/elsql/ElsqlEndpoint.java [hunk 2]
```diff
@@ -139,18 +139,14 @@ private void parseResources(String resourceUri, Predicate<String> predicate) thr
...
-                    try {
-                        final URL url = ResourceHelper.resolveMandatoryResourceAsUrl(getCamelContext().getClassResolver(), path.toString());
-                        list.add(url);
-                    } catch (Exception e) {
-
-                    }
-                });
+        final Iterable<String> it = ObjectHelper.createIterable(resourceUri);
+
+        for(final String path : it) {
```
Comment (davsclaus) [inline components/camel-elsql/src/main/java/org/apache/camel/component/elsql/ElsqlEndpoint.java:144]: Can you fix CS, eg run mvn compile -P sourcecheck

**Comment by davsclaus** [comment]
@zhfeng can you make sure your PRs pass checkstyle, eg from maven run 

mvn compile -P sourcecheck

on the modules you fix. Also make sure to test them first also via mvn test

**Comment by davsclaus** [comment]
Thanks @zhfeng keep em coming ;)

**Pull request merged**